packages/webapp/src/utils/api.tsx

```diff
@@ -174,3 +174,11 @@ export function useGetHmacAPI(env: string) {
         }
     };
 }
+
+export class APIError extends Error {
+    details;
+    constructor(details: { res: Response; json: Record<string, any> | ApiError<any> }) {
+        super('err');
```

Comment:
not sure where you are going with this error subclass. Should the default message be more explicit, even something like `api_error`?